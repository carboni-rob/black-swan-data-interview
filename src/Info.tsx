import React from "react";

const Info = (): JSX.Element => {
  return <div></div>;
};

export default Info;
