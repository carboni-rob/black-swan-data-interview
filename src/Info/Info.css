nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1vmax;
  border-bottom: 2px solid black;
}

nav .mainPageButton {
  width: 120px;
}

.infoPageLogo {
  width: 160px;
}

main {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  background-color: black;
}

.card {
  width: 300px;
  height: 500px;
  margin: 3vmax;
  border: 2px solid white;
  border-top-left-radius: 50px;
  border-bottom-right-radius: 50px;
  box-shadow: 0px 10px 30px 3px rgba(255,255,255,1);
}