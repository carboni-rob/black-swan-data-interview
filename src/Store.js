import React from "react";

export const reducer = (state, action) => {
  switch (action.type) {
    case "resetState":
      return {
        userRepos: [],
        userData: {}
      };
    case "setUserRepos":
      return {
        ...state,
        userRepos: action.payload
      };

    case "setUserData":
      return {
        ...state,
        userData: action.payload
      };

    default:
      return state;
  }
};

export const Context = React.createContext();
