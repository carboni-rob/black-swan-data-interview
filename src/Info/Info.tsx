import React, { useState, useContext } from "react";
import { Redirect } from "react-router-dom";
import { Button } from "antd";
import { Context } from "../Store";
import "./Info.css";
import bsdLogo from "../assets/black_swan_logo.png";

const Info = (): JSX.Element => {
  const { store, dispatch } = useContext(Context);
  const { userRepos, userData } = store;
  const [navigateToSearch, setNavigateToSearch] = useState(false);

  const handleBackButtonClick = (): void => {
    dispatch({ type: "resetState" });
    setNavigateToSearch(true);
  };

  return (
    <div>
      {navigateToSearch && <Redirect to="/" />}
      <nav>
        <Button
          className="mainPageButton"
          type="primary"
          size="large"
          shape="round"
          icon="left-circle"
          onClick={(): void => handleBackButtonClick()}
        >
          BACK
        </Button>
        <img
          src={bsdLogo}
          alt="Black Swan Data logo"
          className="infoPageLogo"
        />
        <div />
      </nav>
      <main>
        {userData && (
          <div className="card">
            <h2>User Info:</h2>
            <img
              src={userData.user.avatar_url}
              alt="User Avatar"
              className="userAvatar"
            />
            <h3>User name: {userData.user.name || "not set"}</h3>
            <h3>Company: {userData.user.company || "not set"}</h3>
            <h3>Email: {userData.user.email || "not set"}</h3>
            <h3>Followers: {userData.user.followers}</h3>
            <h3>Following: {userData.user.following}</h3>
          </div>
        )}
        {userRepos && (
          <div className="card">
            <h2>User Repositories: {userRepos.length}</h2>
            <div className="cardContent">
              {userRepos.map(
                (repo: any): JSX.Element => {
                  return (
                    <a href={repo.clone_url} key={repo.id}>
                      {repo.name}
                    </a>
                  );
                }
              )}
            </div>
          </div>
        )}
        {userData && (
          <div className="card">
            <h2>User Organizations: {userData.orgs.length}</h2>
            <div className="cardContent">
              {userData.orgs.map(
                (org: any): JSX.Element => {
                  return (
                    <a href={org.url} key={org.id}>
                      <img alt="" src={org.avatar_url} className="orgAvatar" />
                      {org.login}
                    </a>
                  );
                }
              )}
            </div>
          </div>
        )}
      </main>
    </div>
  );
};

export default Info;
