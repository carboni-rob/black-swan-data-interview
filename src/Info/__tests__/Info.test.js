/* eslint-disable @typescript-eslint/camelcase */
import React from "react";
import { render } from "@testing-library/react";
import { Context } from "../../Store";
import Info from "../Info";

const store = {
  userData: {
    user: {
      name: "Roberto",
      company: "Black Swan Data",
      email: "carboni@robertocarboni.it",
      followers: 100,
      following: 100,
      avatar_url: "fake/avatar/path"
    },
    orgs: [
      {
        id: 1,
        login: "test1",
        url: "test1/orgs/url",
        avatar_url: "test1/avatar/url"
      }
    ]
  },
  userRepos: [
    {
      id: 1,
      name: "test1",
      clone_url: "test1/repos/url"
    }
  ]
};

describe("<Info />", () => {
  const dispatch = jest.fn();
  const tree = (
    <Context.Provider value={{ store, dispatch }}>
      <Info />
    </Context.Provider>
  );

  it("should match last snapshot", () => {
    const { container } = render(tree);

    expect(container.firstChild).toMatchSnapshot();
  });
});
