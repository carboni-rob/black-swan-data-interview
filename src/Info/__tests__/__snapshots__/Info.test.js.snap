// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Info /> should match last snapshot 1`] = `
<div>
  <nav>
    <button
      class="ant-btn mainPageButton ant-btn-primary ant-btn-round ant-btn-lg"
      type="button"
    >
      <i
        aria-label="icon: left-circle"
        class="anticon anticon-left-circle"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="left-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M603.3 327.5l-246 178a7.95 7.95 0 0 0 0 12.9l246 178c5.3 3.8 12.7 0 12.7-6.5V643c0-10.2-4.9-19.9-13.2-25.9L457.4 512l145.4-105.2c8.3-6 13.2-15.6 13.2-25.9V334c0-6.5-7.4-10.3-12.7-6.5z"
          />
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
        </svg>
      </i>
      <span>
        BACK
      </span>
    </button>
    <img
      alt="Black Swan Data logo"
      class="infoPageLogo"
      src="black_swan_logo.png"
    />
    <div />
  </nav>
  <main>
    <div
      class="card"
    >
      <h2>
        User Info:
      </h2>
      <img
        alt="User Avatar"
        class="userAvatar"
        src="fake/avatar/path"
      />
      <h3>
        User name: 
        Roberto
      </h3>
      <h3>
        Company: 
        Black Swan Data
      </h3>
      <h3>
        Email: 
        carboni@robertocarboni.it
      </h3>
      <h3>
        Followers: 
        100
      </h3>
      <h3>
        Following: 
        100
      </h3>
    </div>
    <div
      class="card"
    >
      <h2>
        User Repositories: 
        1
      </h2>
      <div
        class="cardContent"
      >
        <a
          href="test1/repos/url"
        >
          test1
        </a>
      </div>
    </div>
    <div
      class="card"
    >
      <h2>
        User Organizations: 
        1
      </h2>
      <div
        class="cardContent"
      >
        <a
          href="test1/orgs/url"
        >
          <img
            alt="1 Organization Avatar"
            class="orgAvatar"
            src="test1/avatar/url"
          />
          test1
        </a>
      </div>
    </div>
  </main>
</div>
`;
